optimade/adapters/structures/pymatgen.py

```diff
@@ -0,0 +1,89 @@
+from typing import Union, Dict, List
+
+from optimade.models import Species as OptimadeStructureSpecies
+from optimade.models import StructureResource as OptimadeStructure
+from optimade.models.structures import Periodicity
+
+try:
+    from pymatgen import Structure, Molecule
+
+except (ImportError, ModuleNotFoundError):
+    from warnings import warn
+
+    Structure = None
+    Molecule = None
+    PYMATGEN_NOT_FOUND = "Pymatgen not found, cannot convert structure to a pymatgen Structure or Molecule"
+
+
+__all__ = ("get_pymatgen_structure",)
+
+
+def get_pymatgen_structure(
```

Comment:
I guess the consistent name here would be `get_pymatgen ?

Reply:
I see you have a different convention for `ase` (`get_ase_atoms`)  vs pdb (`get_pdb`)

Reply:
Agreed with `get_pymatgen`. Maybe even if we should follow the convetion for the others (i.e., AiiDA and ASE) it should be `get_pymatgen_structure_molecule`. I like the shorter, more concise version, better though.

Yup. I think I'll change `get_ase_atoms` to `get_ase`.

For AiIDA, I think it's still fine to keep the extra explicit mentioning of the Node in question, since it can be expanded to include `get_aiida_cifnode` or something as well...?

Reply:
> I see you have a different convention for `ase` (`get_ase_atoms`) vs pdb (`get_pdb`)

For all of these, the function name that converts may differ from the actual format to be specified in the `get_` attribute. So one must indeed specify `get_pymatgen` to get a pymatgen Structure or Molecule, as well as one must specify `get_ase` to get an ASE Atoms instance.
But you want the conversion function to reflect the attribute name?